yolox/tracker/byte_tracker.py

```diff
@@ -143,20 +150,29 @@ def __repr__(self):
 
 
 class BYTETracker(object):
-    def __init__(self, args, frame_rate=30):
+    def __init__(self, args, feature_extractor, global_stracks:List[STrack]=[], frame_rate=30):###<----- added feature_extractor & global_stracks attribute to constructor
```

Comment:
if I'm not wrong, `feature_extractor` should be feature extraction model right?

Reply:
## Feature Extractor in BYTETracker

Yes, the `feature_extractor` is passed as an argument when initializing BYTETracker. This extractor is responsible for generating features from detected objects, which are then used for tracking purposes.

The specific `FeatureExtractor` class I use comes from the `deep-person-reid` repository:

[https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/utils/feature_extractor.py](https://github.com/KaiyangZhou/deep-person-reid/blob/master/torchreid/utils/feature_extractor.py)
![image](https://github.com/ifzhang/ByteTrack/assets/91583175/26bf1929-d003-4fe3-8aee-bc82f9315862)

Reply:
Also, keep in mind that the  update method of BYTETracker class in original ByteTrack repo ([here: https://github.com/ifzhang/ByteTrack.git]) does not use feature extraction for tracking. It simply use kalman filter to predict the position of bounding boxes in current frame based on position & velocity data associated with previous frame objects. But there is one drawback with it- kalman filter cannot predict the box for a person lost for more than certain no. of frames (30 by default). So if person is lost for more than 30 frames, it simply ignores that person and if that person re-appears, assigns a new id to him/her. This is not right as it's the same person re-appearing. So, I try to resolve that by storing features of every unique person detected so far. This way, if person is lost for more than buffer-size(30 frames-default) and reappears, I compare the features of that new-appeared person with previously stored ones and infer whether it's same person or not.

Reply:
LGTM!